### ISSUE TITLE
Support Unicode in kNN nearest neighbor clusterers

### ISSUE THREAD

**Issue by tfmorris** [OpenRefine/OpenRefine#3616]
The current tokenizer which is used to normalize & tokenize text isn't Unicode aware. We can either enhance it or switch to a different tokenizer. The SecondString tokenizer provides functionality, with associated overhead, that we don't use, so we may want to switch to something simpler that returns Java `String`s as opposed to `Token` objects.

This should include updating the regexes used for punctuation and control removal as well as the space compression to be Unicode aware, as well as using Unicode aware case folding (ie lowercase). It probably should also do NFKC normalization and perhaps diacritic folding, although the last isn't clear to me.

**Comment by wetneb** [comment]
Beyond aligning them, would it make sense to have them computed by the same method even? It would be great to avoid as much duplication as we can.

**Comment by tfmorris** [comment]
> would it make sense to have them computed by the same method even? It would be great to avoid as much duplication as we can.

Of course, provided structural considerations allow (they are currently in separate components).
